webapp/params.go

```diff
@@ -77,10 +80,14 @@ func ParseBrowsersParam(r *http.Request) (browsers []string, err error) {
 	// Otherwise filter to valid browser names.
 	for i := 0; i < len(browsers); {
 		if !IsBrowserName(browsers[i]) {
-			// 'Remove' browser by switching to end and cropping.
-			browsers[len(browsers)-1], browsers[i] = browsers[i], browsers[len(browsers)-1]
-			browsers = browsers[:len(browsers)-1]
-			continue
+			if browsers[i] == "" {
+				// 'Remove' empty browser by switching to end and cropping.
```

Comment:
Vein curiosity: Is this a golang idiom that yields better performance (because you're not splicing the slice)?

Reply:
It's not for efficiency; you just can't easily remove an item... I think `append(browsers[0:i], ...browsers[i+1:])` would actually do the same in one line, but definitely less efficiently.

Optimizing for a list of size 4: 👍